tests/loom.rs

```diff
@@ -0,0 +1,47 @@
+#[cfg(loom)]
+#[cfg(test)]
+mod loom_tests {
+    use left_right::{Absorb, ReadGuard};
+    use loom::sync::atomic::AtomicUsize;
+    use loom::sync::atomic::Ordering::{Acquire, Relaxed, Release};
+    use loom::sync::Arc;
+    use loom::thread;
+
+    // TODO Would love to not re-define this!
+    struct CounterAddOp(i32);
+
+    impl Absorb<CounterAddOp> for i32 {
+        fn absorb_first(&mut self, operation: &mut CounterAddOp, _: &Self) {
+            *self += operation.0;
+        }
+
+        fn absorb_second(&mut self, operation: CounterAddOp, _: &Self) {
+            *self += operation.0;
+        }
+
+        fn drop_first(self: Box<Self>) {}
+
+        fn sync_with(&mut self, first: &Self) {
+            *self = *first
+        }
+    }
+
+    #[test]
+    fn read_before_publish() {
+        loom::model(|| {
+            let (mut w, r) = left_right::new::<i32, _>();
+            w.append(CounterAddOp(1));
+            w.publish();
+
+            let val = *r.enter().unwrap();
```

Comment:
I think this read should be after the thread spawn but before the join and I should remove the publish from the thread closure?

Reply:
Huh I updated this with 753fd8704bc7959b73249a35abe6c00524c57983 but now, when running loom tests I get:
```
thread 'loom_tests::read_before_publish' panicked at 'Model exeeded maximum number of branches. 
```
which I think is something I ran into [last time](https://github.com/jonhoo/left-right/pull/62#discussion_r468213451) ... not sure what's going on here 🤔

Reply:
The updated code looks right to me. What value are you running with for `MAX_PREEMPTIONS`?

Reply:
Hmmm so I locally added the `loom::thread::yield_now` (good call!) but I'm realizing that this test may not be deterministic - if the write handle is dropped before the read happens, the reader will return `None`. Is that right?

Reply:
So I think the more safe test is to read the value in the thread and do the publish between the spawn and join. Then, we can ensure the write handle stays alive long enough to validate the test by using it after the join/assert/something?

Reply:
Made those adjustments in b7442f68147bd7bb505f56615f6e3f3e2489fc6c. Lemme know if that's right! If so, I think the last step is just to update CI